jest.config.js

```diff
@@ -1,19 +1,43 @@
+const isRunningInEsmMode = !!process.env.TEST_ESM
+
+/**
+ * These packages are authored using native ESModules, and so we want to configure jest differently to handle that
+ *
+ * TODO: determine this from "type": "module" in package.json
+ */
+const ESM_PACKAGES = ['content-conformance', 'remark-plugins']
+
+const esmConfig = {
+  extensionsToTreatAsEsm: ['.ts', '.tsx'],
+  transform: {
+    '^.+\\.(t)sx?$': ['@swc/jest', { module: { type: 'es6' } }],
+  },
+}
+
+/**
+ * Exclude ESM-native packages from running when not in ESM mode
+ */
+const ignorePatternForModuleType = isRunningInEsmMode
+  ? `<rootDir>/packages/(?!${ESM_PACKAGES.join('|')}).*/.*`
+  : `<rootDir>/packages/(${ESM_PACKAGES.join('|')})/.*`
+
 module.exports = {
-  preset: 'ts-jest',
   testEnvironment: 'node',
   verbose: true,
-  globals: {
-    'ts-jest': {
-      diagnostics: {
-        warnOnly: true,
-      },
-    },
+  transform: {
+    '^.+\\.(j|t)sx?$': ['@swc/jest'],
```

Comment:
What's the thought behind `@swc/jest`? (This is new to me)

Reply:
I found it much more straight forward to configure than `ts-jest` out of the box, and it's faster.